src/guest_memory.rs

```diff
@@ -295,8 +295,8 @@ pub trait GuestMemory {
         let mut total = 0;
         while let Some(region) = self.find_region(cur) {
             let start = region.to_region_addr(cur).unwrap();
-            let cap = region.len() as usize;
-            let len = std::cmp::min(cap, count - total);
+            let cap = region.len() - start.raw_value();
+            let len = std::cmp::min(cap as usize, count - total);
```

Comment:
There is still a risk that cap is bigger than usize on 32-bit platforms.
How about:
            let cap = region.len() - start.raw_value();
            let len = std::cmp::min(cap, (count - total) as GuestUsize);	   
           match f(total, len as usize, start, region) {

Reply:
Yes, it's better. @bjzhjing can you adjust?

Reply:
Sure, change is done!